### ISSUE TITLE
Another multi-gpu finetuning repository without tokeniser issue.

### ISSUE THREAD

**Issue by cookyecat** [andimarafioti/florence2-finetuning#15]
I run the [florence2-finetuning](https://github.com/andimarafioti/florence2-finetuning) in my machine, but got the same issue on tokeniser as #14. I tried many methods but can not fix.

So I wrote another [florence-2 multi-gpu finetuning project](https://github.com/Coldmooon/florence-2-finetune), especially for multi-gpu training. The codes heavily reference the [Hugging Face blog on fine-tuning Florence-2] (https://huggingface.co/blog/finetune-florence2) blog and single-gpu training codes. I used `Chatgpt`  to expand the multi-GPU training codes and then manually removed bugs. Interestingly, the codes writen by chatgpt is very similary to those given in this repo, but work fine.

**Comment by andimarafioti** [comment]
Wouldn't it have been easier to write a PR to fix the issue here?

**Comment by andimarafioti** [comment]
For what is worth, I'm the main author of the blog you say I'm heavily referencing and I wrote the code first for multi-gpu training and then adapted it to single-gpu for the blog

**Comment by andimarafioti** [comment]
Pushed the tokenizer fix: https://github.com/andimarafioti/florence2-finetuning/commit/262258ef647b83c08e0d574b2cce5f4f4b4c9e95

**Comment by cookyecat** [comment]
> Wouldn't it have been easier to write a PR to fix the issue here?

I thought of that. But I didn't find the right solution to #14 , and codes written by chatgpt have different structure from your codes. If I simply wirte a PR, your code structure will be changed a lot without reasonable explanation I can give.

**Comment by cookyecat** [comment]
> For what is worth, I'm the main author of the blog you say I'm heavily referencing and I wrote the code first for multi-gpu training and then adapted it to single-gpu for the blog

Thanks for your work. I really learn a lot from your blog. 

Actually, I first found and read your blog, and then tried your single-gpu codes. After that, I used Chatgpt to expand your single-gpu codes to multi-gpu codes and fixed bugs by hand. Then, I found your github repository and downloaded your repository.  When I run [distributed_train.py](https://github.com/andimarafioti/florence2-finetuning/blob/main/distributed_train.py), I meet the issue same as #14.
